### ISSUE TITLE
Wrapping `n-button` inside `n-upload` will break `n-button-group` style

### ISSUE THREAD

**Issue by equt** [TuSimple/naive-ui#1102]
### TuSimple/naive-ui version (版本)
2.17.1

### Vue version (Vue 版本)
3.2.9

### Browser and its version (浏览器及其版本)
Chrome(92.0.4515.159)

### System and its version (系统及其版本)
macOS(11.2.3)

### Reappearance link (重现链接)
https://github.com/equt-playground/naive-ui-upload-inside-button-group

### Reappearance steps (重现步骤)
1. Wrap an `n-button` inside an `n-upload` component;
2. Place the wrapped `n-button` inside an `n-button-group`.

### Expected results (期望的结果)
The wrapped button should be viewed as transparent.

### Actual results (实际的结果)
The wrapped button is not treated as a button anymore.

### Remarks (补充说明)
For a live demo, see the [playground](https://playground.equt.services/naive-ui-upload-inside-button-group/#/).

**Comment by 07akioni** [comment]
I've a good idea to make this fixed. I need some time to make it work.

Alse the idea can make file list and trigger into different components.

```ts
<n-upload abstract>
  <n-upload-trigger abstract #="{ handleClick }">
    <n-button @click="handleClick"></n-button>
  </n-upload-trigger>
  <n-modal>
    <n-upload-file-list />
  </n-modal>
</n-upload>
```

**Comment by 07akioni** [comment]
fixed in #1193
